src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Andrews54757
> Do you even know how GC works?
- GC in V8 works by using a "generational collection" implementation. All of GC's memory is stored in `Page`s, which are allocations of 1MB of memory. All new objects and raw values (doubles, strings and arrays) join a collection called `New-space` in the form of pointers. Objects that survive a period of time (by having a pointer pointing to the object) join either `Old-pointer-space`, `Old-data-space` or `Large-object-space`. `Old-pointer-space` are a collection of pointers to objects that probably contain pointers to other objects or raw values. `Old-data-space` contains pointers to raw values. `Large-object-space` contains big objects that have their own 1MB of allocated memory and have a very low chance of being marked as dead. GC uses Mark-sweep and Mark-compact methods to mark dead and alive pointers using a white-grey-black system. Once marked, Mark-sweep records the dead and alive pointers and Mark-compact removes dead object pointers, thereby freeing memory.
- I missed out some parts of it such as pointer types, write barriers, incremental markings and lazy sweeping.
- If you still think that I don't know about GC, explain it.

> V8 is dumb enough.
- Its garbage collection engine is super fast and very reliable.

> And do you know what happens when memory is deallocated? Holes start to appear.
- What "holes"?

> Also, I dont care about your life.
- Neither do I about yours.

> But I bet your resume, if exists, will mostly be empty if you exclude all games.
- We don't have resumes in our country.

> How about you answer my questions? Should I?
- No. I don't know you. You don't know me. I don't want to know you. Betting you don't want to know me too.